File: Question6/CPP/Question6.cpp
#include <bits/stdc++.h>

using namespace std;

string encrypt(string words)
{
    words.erase(std::remove(words.begin(), words.end(), ' '), words.end());
    double esquerte = sqrt(words.length());
    int rows = floor(esquerte);
    int cols = ceil(esquerte);
    if (rows * cols < words.length())
        rows++;
    string return_string = "";

    for (int i = 0; i < rows * cols; i++)
    {
        int iprzecols = i / rows;
        int val = (i % rows) * cols + iprzecols;
        if (val < words.length())
            return_string += words[val];
        if ((i + 1) % rows == 0)
            return_string += " ";
    }

    return return_string;
}

int main()
{
    string words;
    getline(cin, words);

    string result = encrypt(words);

    // Do not remove below line
    cout << result << "\n";
    // Do not print anything after this line

    return 0;
}

File: Question6/Python/Question6.py
import math


# You may change this function parameters
def encrypt(words):
    words.replace(" ", "")
    sqrt = math.sqrt(len(words))
    rows = int(math.floor(sqrt))
    cols = int(math.ceil(sqrt))
    if rows * cols < len(words):
        rows += 1
    return_string = []
    for i in range(0, rows * cols):
        val = ((i % rows) * cols) + (int(i / rows))
        if val < len(words):
            return_string.append(words[val])
        if (i + 1) % rows == 0:
            return_string.append(" ")

    # Participants code will be here
    return "".join(return_string)


def main():
    words = input()

    answer = encrypt(words)

    # Please do not remove the below line.
    print(answer)
    # Do not print anything after this line


if __name__ == '__main__':
    main()
